### PR TITLE
Run on XML pages (version 2)

### PR DESCRIPTION
**Pull request by gdh1995**
This is based on mrmr1993 version, and  fixs some bugs found in PR #1797.

### CODE REVIEW

**Comment by gdh1995** [comment]
Sorry. I'll do `cake test` on my computer. Just a while.

**Comment by gdh1995** [comment]
If better, I can rebase my repo and create a new PR ,since these commit logs are awful.

**Comment by mrmr1993** [comment]
Code LGTM (but haven't tested since I'm on holiday). I'll close #1797.

File: content_scripts/vimium_frontend.coffee [hunk 1]
```diff
@@ -274,23 +274,22 @@ setScrollPosition = ({ scrollX, scrollY }) ->
 #
 # Called from the backend in order to change frame focus.
 #
-window.focusThisFrame = do ->
+setTimeout (->
```
Comment (smblott-github) [inline content_scripts/vimium_frontend.coffee]: @gdh1995.  Thanks for this.

Effectively, you're running this on "nextTick".  How do we know that `createElement` will work then, if it won't work now?  I kinda expected to see `DomUtils.documentReady` here.

(Separately -- and for future reference -- we have `Utils.setTimeout()` and `Utils.nextTick()`.)

**Comment by gdh1995** [comment]
I did such a test: modify `manifest.json` and set `.content_scripts[0].run_at = "document_end"`, and then the old `vimium_frontend.coffee` would not make a XML view page reset into a RSS one. This means that we just need to call every `createElementNS` after `document.readyState` is `"interactive"` (more on [developer.../.../content_scripts](https://developer.chrome.com/extensions/content_scripts)).

Since the main thread parses HTML content and executes scripts serially, it seems that `setTimeout` will work only after the page is ready, but if a server blocks the XML body (e.g. sleeps some seconds before send a next chunk), I do know whether the main thread will be blocked or not.

So, let's use `documentReady` firstly.

**Comment by smblott-github** [comment]
Thanks, @gdh1995.

**Pull request merged**